### PR TITLE
Cleanup unused methods and fields in pkg/policy

### PR DESCRIPTION
**Pull request by aanm**
Read per commit

### CODE REVIEW

**Review by christarazi — APPROVED** [review]
Nice. I thought `golangci-lint` would have caught these things?

**Comment by aanm** [comment]
test-me-please

File: pkg/policy/distillery.go [hunk 1]
```diff
@@ -64,24 +64,21 @@ func (cache *PolicyCache) GetSelectorCache() *SelectorCache {
 // lookupOrCreate adds the specified Identity to the policy cache, with a reference
 // from the specified Endpoint, then returns the threadsafe copy of the policy
 // and whether policy has been computed for this identity.
```
Comment (aditighag) [inline pkg/policy/distillery.go]: Update the description for returned fields.

File: pkg/policy/distillery.go [hunk 2]
```diff
@@ -64,24 +64,21 @@ func (cache *PolicyCache) GetSelectorCache() *SelectorCache {
...
-	if cip != nil {
-		return cip, cip.getPolicy().Revision > 0
-	}
-	return nil, false
+	return cip
 }
 
 // insert adds the specified Identity to the policy cache, with a reference
 // from the specified Endpoint, then returns the threadsafe copy of the policy
 // and whether policy has been computed for this identity.
```
Comment (aditighag) [inline pkg/policy/distillery.go]: Same as above.

**Review by aditighag — CHANGES_REQUESTED** [review]

**Comment by aanm** [comment]
test-me-please

**Review by aditighag — APPROVED** [review]

**Review by joestringer — APPROVED** [review]

**Comment by sayboras** [comment]
> Nice. I thought `golangci-lint` would have caught these things?

I think golangci-lint is only able to catch un-used private methods/variables, it's hard to catch public ones in `pkg/` , as these can be used as depedency.

**Comment by aanm** [comment]
retest-netnext

**Comment by aanm** [comment]
retest-net-next

**Comment by jrajahalme** [comment]
Other backports depended on these so backported to v1.8.

**Pull request merged**